website/src/theme/DocSidebar/index.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import React, { useState } from "react";
+import clsx from "clsx";
+import {
+  useThemeConfig,
+  useAnnouncementBar,
+  MobileSecondaryMenuFiller,
+  ThemeClassNames,
+  useScrollPosition,
+} from "@docusaurus/theme-common";
+import useWindowSize from "@theme/hooks/useWindowSize";
+import Logo from "@theme/Logo";
+import IconArrow from "@theme/IconArrow";
+import { translate } from "@docusaurus/Translate";
+import { DocSidebarItems } from "@theme/DocSidebarItem";
+import styles from "./styles.module.css";
+
+function useShowAnnouncementBar() {
+  const { isActive } = useAnnouncementBar();
+  const [showAnnouncementBar, setShowAnnouncementBar] = useState(isActive);
+  useScrollPosition(
+    ({ scrollY }) => {
+      if (isActive) {
+        setShowAnnouncementBar(scrollY === 0);
+      }
+    },
+    [isActive]
+  );
+  return isActive && showAnnouncementBar;
+}
+
+function HideableSidebarButton({ onClick }) {
+  return (
+    <button
+      type="button"
+      title={translate({
+        id: "theme.docs.sidebar.collapseButtonTitle",
+        message: "Collapse sidebar",
+        description: "The title attribute for collapse button of doc sidebar",
+      })}
+      aria-label={translate({
+        id: "theme.docs.sidebar.collapseButtonAriaLabel",
+        message: "Collapse sidebar",
+        description: "The title attribute for collapse button of doc sidebar",
+      })}
+      className={clsx("button button--secondary button--outline", styles.collapseSidebarButton)}
+      onClick={onClick}>
+      <IconArrow className={styles.collapseSidebarButtonIcon} />
+    </button>
+  );
+}
+
+function DocSidebarDesktop({ path, sidebar, onCollapse, isHidden }) {
+  const showAnnouncementBar = useShowAnnouncementBar();
+  const {
+    navbar: { hideOnScroll },
+    hideableSidebar,
+  } = useThemeConfig();
+  return (
+    <div
+      className={clsx(styles.sidebar, {
+        [styles.sidebarWithHideableNavbar]: hideOnScroll,
+        [styles.sidebarHidden]: isHidden,
+      })}>
+      {hideOnScroll && <Logo tabIndex={-1} className={styles.sidebarLogo} />}
+      <nav
+        className={clsx("menu", styles.menu, {
```

Comment:
nit: wdyt about adding `thin-sidebar` (from the original nav) to this element?

On non-macOS it looks a bit sleeker imo and conserves horizontal space.

Before

![image](https://user-images.githubusercontent.com/2414826/216626754-02468560-2d4c-43a4-ad96-df477236a276.png)

After

![image](https://user-images.githubusercontent.com/2414826/216626522-8af48bef-a024-44b6-9d23-76fe75dfc038.png)

Reply:
agh, auto-merge strikes again. I'll send a PR